packages/ibm-products/src/components/TagOverflow/TagOverflow.js

```diff
@@ -0,0 +1,266 @@
+
+      const ItemComp = () =>
+        React.createElement(itemTemplate, {
+          ref: itemRef,
+          ...templateProps,
+          className,
+        });
+
+      return <ItemComp></ItemComp>;
+    };
+
+    ItemComponent.propTypes = {
+      templateProps: PropTypes.shape({
+        id: PropTypes.string.isRequired,
+        className: PropTypes.string,
+      }).isRequired,
+    };
+
+    useEffect(() => {
+      let visibleItemsArr = getVisibleItems();
+
+      if (maxVisible && maxVisible < visibleItemsArr.length) {
+        visibleItemsArr = visibleItemsArr?.slice(0, maxVisible);
+      }
+
+      const hiddenItems = items?.slice(visibleItemsArr.length);
+      const overflowItemsArr = hiddenItems?.map((item) => {
+        return { type: tagType, label: item.label };
+      });
+
+      setVisibleItems(visibleItemsArr);
+      setOverflowItems(overflowItemsArr);
+    }, [containerWidth, items, maxVisible, getVisibleItems]);
+
+    return (
+      <div
+        {
+          // Pass through any other property values as HTML attributes.
+          ...rest
+        }
+        className={cx(
+          blockClass, // Apply the block class to the main HTML element
+          className, // Apply any supplied class names to the main HTML element.
+          // example: `${blockClass}__template-string-class-${kind}-n-${size}`,
+          {
+            // switched classes dependant on props or state
+            // example: [`${blockClass}__here-if-small`]: size === 'sm',
+          }
+        )}
+        ref={containerRef}
+        role="main"
+        {...getDevtoolsProps(componentName)}
+      >
+        {!!visibleItems?.length &&
+          visibleItems.map((item) => {
+            // render custom components
+            if (itemTemplate) {
+              return (
+                <ItemComponent
+                  templateProps={item}
+                  key={item.id}
+                ></ItemComponent>
+              );
+            } else {
+              // if there is no template prop, then render items as Tags
+              return (
+                <div
+                  ref={(node) => itemRefHandler(item.id, node)}
+                  key={item.id}
+                >
+                  <Tooltip align="bottom" label={item.label}>
+                    <Tag className={`${blockClass}__item--tag`} type={tagType}>
+                      {item.label}
+                    </Tag>
+                  </Tooltip>
+                </div>
+              );
+            }
+          })}
+        <span className={`${blockClass}__indicator`} ref={overflowRef}>
+          {!!overflowItems?.length && (
+            <TagSet
+              tags={overflowItems}
+              allTagsModalTitle="All tags"
+              containingElementRef={overflowRef}
+              allTagsModalSearchLabel="Search all tags"
+              allTagsModalSearchPlaceholderText="Search all tags"
+              showAllTagsLabel="Show all tags"
+            ></TagSet>
```

Comment:
for any element declaration that doesn't have a child you can just use the `<TagSet />` notation